stacks_queues/queue.py

```diff
@@ -15,47 +15,68 @@ def __init__(self):
         self.front = -1
         self.rear = -1
         self.size = 0
-      
 
     def enqueue(self, element):
         """ Adds an element to the Queue
             Raises a QueueFullException if all elements
             In the store are occupied
             returns None
         """
-        pass
+        if self.size == self.buffer_size:
+            raise QueueFullException("Queue full")
+
+        if self.size == 0:
+            self.front = 0
+            self.rear = 0
+
+        self.store[self.rear] = element
+        self.rear = (self.rear + 1) % self.buffer_size
+        self.size += 1
 
     def dequeue(self):
         """ Removes and returns an element from the Queue
             Raises a QueueEmptyException if 
             The Queue is empty.
         """
-        pass
+        if self.empty():
+            raise QueueEmptyException("Queue is empty")
+        val = self.store[self.front]
+        self.store[self.front] = None
+        self.front = (self.front + 1) % self.buffer_size
+        self.size -= 1
+        return val
+        
 
     def front(self):
         """ Returns an element from the front
             of the Queue and None if the Queue
             is empty.  Does not remove anything.
         """
-        pass
+        return self.store[self.front]
         
 
     def size(self):
         """ Returns the number of elements in
             The Queue
         """
-        pass
+        return self.size
 
     def empty(self):
         """ Returns True if the Queue is empty
             And False otherwise.
         """
-        pass
+        return self.size == 0
 
     def __str__(self):
         """ Returns the Queue in String form like:
             [3, 4, 7]
             Starting with the front of the Queue and
             ending with the rear of the Queue.
         """
-        pass
+        arr = []
+        for i in range(self.size):
+            index = (self.front + i) % self.buffer_size
+            arr.append(str(self.store[index]))
+        
+        return "[" + ", ".join(arr) + "]"
```

Comment:
👀 Stringifying an array will include the brackets!
```suggestion
        return str(arr)